File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/AnyofWithOneEmptySchemaTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class AnyofWithOneEmptySchemaTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().format().build());

    @Test
    public void testNumberIsValidPasses() throws ValidationException {
        // number is valid
        final var schema = AnyofWithOneEmptySchema.AnyofWithOneEmptySchema1.getInstance();
        schema.validate(
            123,
            configuration
        );
    }

    @Test
    public void testStringIsValidPasses() throws ValidationException {
        // string is valid
        final var schema = AnyofWithOneEmptySchema.AnyofWithOneEmptySchema1.getInstance();
        schema.validate(
            "foo",
            configuration
        );
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/AnyofWithOneEmptySchemaTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class AnyofWithOneEmptySchemaTest {
    companion object {
        val configuration = SchemaConfiguration(JsonSchemaKeywordFlags.Builder().format().build())
    }

    @Test
    fun testNumberIsValidPasses() {
        // number is valid
        val schema = AnyofWithOneEmptySchema.AnyofWithOneEmptySchema1.getInstance()
        schema.validate(
            123,
            configuration
        )
    }

    @Test
    fun testStringIsValidPasses() {
        // string is valid
        val schema = AnyofWithOneEmptySchema.AnyofWithOneEmptySchema1.getInstance()
        schema.validate(
            "foo",
            configuration
        )
    }
}
